test/http_content_length_test.dart

```diff
@@ -0,0 +1,240 @@
+      if (useHeader) {
+        response.contentLength = 2;
+        response.headers.chunkedTransferEncoding = true;
+      } else {
+        response.headers.set("content-length", 2);
+        response.headers.set("transfer-encoding", "chunked");
+      }
+      request.listen((d) {}, onDone: () {
+        response.write("x");
+        try {
+          response.headers.chunkedTransferEncoding = false;
+          fail("Expected HttpException");
+        } catch (e) {
+          expect(e is HttpException, isTrue);
+        }
+        response.write("x");
+        response.write("x");
+        response.close();
+        response.write("x");
+      });
+    }, onError: (e, trace) {
+      String msg = "Unexpected error $e";
+      if (trace != null) msg += "\nStackTrace: $trace";
+      fail(msg);
+    });
+
+    int count = 0;
+    HttpClient client = new HttpClient();
+    for (int i = 0; i < totalConnections; i++) {
+      client.get("127.0.0.1", server.port, "/").then((request) {
+        if (useHeader) {
+          request.contentLength = 2;
+          request.headers.chunkedTransferEncoding = true;
+        } else {
+          request.headers.add(HttpHeaders.CONTENT_LENGTH, "2");
+          request.headers.set(HttpHeaders.TRANSFER_ENCODING, "chunked");
+        }
+        request.write("x");
+        try {
+          request.headers.chunkedTransferEncoding = false;
+          fail("Expected HttpException");
+        } catch (e) {
+          expect(e is HttpException, isTrue);
+        }
+        request.write("x");
+        request.write("x");
+        return request.close();
+      }).then((response) {
+        expect(response.headers.value('content-length'), isNull);
+        expect(-1, equals(response.contentLength));
+        response.listen((d) {}, onDone: () {
+          if (++count == totalConnections) {
+            client.close();
+            server.close();
+          }
+        });
+      }).catchError((e, trace) {
+        String msg = "Unexpected error $e";
+        if (trace != null) msg += "\nStackTrace: $trace";
+        fail(msg);
+      });
+    }
+  });
+}
+
+void testSetContentLength() {
+  HttpServer.bind("127.0.0.1", 0).then((server) {
+    server.listen((HttpRequest request) {
+      var response = request.response;
+      expect(response.headers.value('content-length'), isNull);
+      expect(-1, equals(response.contentLength));
+      response.headers.set("content-length", 3);
+      expect("3", equals(response.headers.value('content-length')));
+      expect(3, equals(response.contentLength));
+      response.write("xxx");
+      response.close();
+    });
+
+    var client = new HttpClient();
+    client
+        .get("127.0.0.1", server.port, "/")
+        .then((request) => request.close())
+        .then((response) {
+      response.listen((_) {}, onDone: () {
+        client.close();
+        server.close();
+      });
+    });
+  });
+}
+
+void main() {
+  testNoBody(5, false);
```

Comment:
These should be converted to package:test format. E.g.:

test('NoBody', () => testNoBody(5, false));

See the other tests in this directory.

Reply:
Done.